mantidimaging/eyes_tests/base_eyes.py

```diff
@@ -27,35 +27,43 @@
 # APPLITOOLS_BATCH_ID will be set by Github actions to the commit SHA, or a random UUID for individual developer
 # execution
 
-APPLITOOLS_BATCH_ID = os.getenv("APPLITOOLS_BATCH_ID")
-if APPLITOOLS_BATCH_ID is None:
-    APPLITOOLS_BATCH_ID = str(uuid4())
-
+APPLITOOLS_BATCH_ID = os.getenv("APPLITOOLS_BATCH_ID") or str(uuid4())
 API_KEY_PRESENT = os.getenv("APPLITOOLS_API_KEY")
+TEST_NAME = os.getenv("GITHUB_BRANCH_NAME") or f"{getpass.getuser()}'s Local Test"
 
-TEST_NAME = os.getenv("GITHUB_BRANCH_NAME")
-if TEST_NAME is None:
-    TEST_NAME = f"{getpass.getuser()}'s Local Test"
-
-LOAD_SAMPLE = str(Path.home()) + "/mantidimaging-data/ISIS/IMAT/IMAT00010675/Tomo/IMAT_Flower_Tomo_000000.tif"
-LOAD_SAMPLE_MISSING_MESSAGE = """Data not present, please clone to your home directory e.g.
-git clone https://github.com/mantidproject/mantidimaging-data.git"""
+env_data_path = os.getenv("MANTIDIMAGING_DATA_DIR")
+POTENTIAL_DATA_DIRS = []
 
-NEXUS_SAMPLE = str(
-    Path.home()) + "/mantidimaging-data/Diamond/i13/AKingUVA_7050wSSwire_InSitu_95RH_2MMgCl2_p4ul_p4h/24737.nxs"
+if env_data_path:
+    POTENTIAL_DATA_DIRS.append(Path(env_data_path))
+POTENTIAL_DATA_DIRS.append(Path(__file__).resolve().parent.parent / "mantidimaging-data")
+POTENTIAL_DATA_DIRS.append(Path.home() / "mantidimaging-data")
 
-APPLITOOLS_IMAGE_DIR = os.getenv("APPLITOOLS_IMAGE_DIR")
-if APPLITOOLS_IMAGE_DIR is None:
-    APPLITOOLS_IMAGE_DIR = mkdtemp(prefix="mantid_image_eyes_")
+for candidate in POTENTIAL_DATA_DIRS:
+    if candidate.is_dir():
+        DATA_ROOT = candidate
+        break
```

Comment:
We can be a bit more pythonic here where instead of checking `POTENTIAL_DATA_DIRS` in order, and using the first directory that exists (which is fine), we could use a generator expression with `next()`:
```Python
DATA_ROOT = next(
    (candidate for candidate in POTENTIAL_DATA_DIRS if candidate.is_dir()),
    None
)
if DATA_ROOT is None:
    raise RuntimeError(...)
```